### PR TITLE
Add support for Google Shared VPC Networks (XPN)

### PR DESCRIPTION
**Pull request by zbikmarc**
[Shared VPC](https://cloud.google.com/compute/docs/shared-vpc/) network was previously know as Cross-Project Networking (XPN).

This PR adds support to provision instances into shared networks and subnetworks.

### CODE REVIEW

**Comment by zbikmarc** [comment]
Thank you @robbkidd

File: lib/kitchen/driver/gce_version.rb [hunk 1]
```diff
@@ -18,6 +18,6 @@
 
 module Kitchen
   module Driver
-    GCE_VERSION = "1.2.0".freeze
+    GCE_VERSION = "1.2.1".freeze
```
Comment (robbkidd) [inline lib/kitchen/driver/gce_version.rb]: @zbikmarc Can you revert this change? We'll gather one or more fixes on master and then do a version bump as part of a release.

Comment (zbikmarc) [reply lib/kitchen/driver/gce_version.rb]: Done

**Comment by robbkidd** [comment]
@zbikmarc Can you update the README to include these additions to config options with some guidance on when to use them? Also, squashing the commits into one informative commit about the changes would be helpful for future us!

**Comment by zbikmarc** [comment]
I will update this tomorrow.
You are not doing squash and merge? oO

File: lib/kitchen/driver/gce.rb [hunk 2]
```diff
@@ -396,21 +407,20 @@ def env_user
 
       def instance_network_interfaces
         interface                = Google::Apis::ComputeV1::NetworkInterface.new
-        interface.network        = network_url
+        interface.network        = network_url if config[:subnet_project].nil?
```
[the remark this answers is not in the document]
Comment (zbikmarc) [reply lib/kitchen/driver/gce.rb]: As far as I remember you need to specify either network or subnetwork, not both of them.
[This](https://github.com/google/google-api-go-client/blob/master/compute/v1/compute-api.json#L5223) might help, same regarding to [Terraform documentation](https://www.terraform.io/docs/providers/google/r/compute_instance.html#subnetwork). Generally subnetwork is more specific then network.

[the remark this answers is not in the document]
Comment (robbkidd) [reply lib/kitchen/driver/gce.rb]: > Either `network` or `subnetwork` must be provided.

A-ha. From the terraform documentation. This is not clear from the Google API documentation.

OK. So my interpretation of this change is that, if all of `network`, `network_project`, `subnet`, and `subnet_project` are given, the specified `subnet` in the `subnet_project` wins. Is that right? And the right behavior?

[the remark this answers is not in the document]
Comment (zbikmarc) [reply lib/kitchen/driver/gce.rb]: Yes, you are right.

**Comment by err0** [comment]
@robbkidd is this mergeable with that understanding?  I would love to use this fix.

**Comment by robbkidd** [comment]
Released in kitchen-google 1.4.0

**Comment by robbkidd** [comment]
Thanks for your patience on this one, folks. I'm super interested in feedback around how this is working for y'all.

**Comment by seanmalloy** [comment]
@robbkidd we have been using this patch internally for several months. It works very well. I also just tested out the official 1.4.0 with this feature and I can also confirm that it is working.

Thanks!
